lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -95,6 +95,179 @@ def self.persist_buffers!
       true #TODO: More useful return value?
     end
 
+    # Check conflicts for Configuration Type Role Updates
+    def self.conflicts_for_operables(operable_ids, crs_ids)
+      descendants_query = <<-SQL
+        WITH RECURSIVE descendants AS (
+          SELECT parent_id, child_id, parent_id as root_node
+          FROM assignments
+          WHERE parent_id IN (#{operable_ids.join(", ")})
+          UNION ALL
+          SELECT a.parent_id, a.child_id, root_node
+          FROM assignments a
+            INNER JOIN descendants d 
+            ON a.parent_id = d.child_id
+          WHERE a.parent_id = d.child_id
+        )
+        SELECT root_node, ra.operator_uri, crs.id
+        FROM descendants d
+          JOIN policy_element_associations AS pea
+          ON pea.object_attribute_id = d.child_id
+            OR pea.object_attribute_id = d.parent_id
+          JOIN policy_elements as ra
```

Comment:
I think I agree with @fpcyan that these aliases should be more semantic, at least for cases like this where the logic of the join depends on whether it's a role assignment, operation set, etc